_drafts/armv7_mpu.md

```diff
@@ -0,0 +1,376 @@
+---
+title: "Catching Bugs and Preventing Security Exploits with the ARM MPU"
+author: chris
+---
+
+<!-- excerpt start -->
+
+On a computer when when an application dereferences a NULL pointer or tries to access kernel memory we can catch it pretty easily thanks to the features provided by the [MMU](https://en.wikipedia.org/wiki/Memory_management_unit#Benefits) on the chip.
+
+For embedded development on MCUs, we aren't always so lucky. A bad memory access from a task responsible for reading an accelerometer could corrupt a buffer being used elsewhere in the system to update a display!
+
+Fortunately, not all hope is lost. Many ARM MCUs implement an optional unit, known as the **Memory Protection Unit (MPU)**, which lets you safeguard how regions of memory are accessed.
+In this article, we will deep dive into the unit and walk through a few practical examples of how it can be used to prevent bad memory accesses and security exploits on _Cortex-M_ devices.
+
+<!-- excerpt end -->
+
+_Like Interrupt? [Subscribe](http://eepurl.com/gpRedv) to get our latest
+posts straight to your mailbox_
+
```

Comment:
This article is long enough that it could benefit from a table of contents

Reply:
hrm, I think it gets a little bit long 
<img width="559" alt="image" src="https://user-images.githubusercontent.com/7494333/61166974-60440200-a505-11e9-8767-783bf8df9972.png">

There's probably a way to only have the TOC show headers out to a certain level though I'm not sure how exactly to do that. I was looking at the info [here])(https://blog.webjeda.com/jekyll-toc/)

Reply:
figured out how to remove some of the sub-headings so I added the TOC!